src/test/regress/expected/rpt_optimizer.out

```diff
@@ -953,15 +953,81 @@ ___________
+               Output: t_replicate_src.i, (int4(nextval('t_replicate_dst_id_seq'::regclass)))
+               ->  Seq Scan on rpt.t_replicate_src
+                     Output: t_replicate_src.i, int4(nextval('t_replicate_dst_id_seq'::regclass))
+ Optimizer: GPORCA
+ Settings: enable_seqscan = 'off'
+(9 rows)
+
+explain (costs off, verbose) with s as (select i from t_replicate_src group by i having random() > 0) insert into t_replicate_dst (i) select i from s;
+                                       QUERY PLAN                                       
+----------------------------------------------------------------------------------------
+ Insert on rpt.t_replicate_dst
+   ->  Broadcast Motion 1:3  (slice1; segments: 1)
+         Output: ((nextval('t_replicate_dst_id_seq'::regclass))::integer), "*SELECT*".i
+         ->  Subquery Scan on "*SELECT*"
+               Output: nextval('t_replicate_dst_id_seq'::regclass), "*SELECT*".i
+               ->  Result
+                     Output: t_replicate_src.i
+                     ->  GroupAggregate
+                           Output: t_replicate_src.i
+                           Group Key: t_replicate_src.i
+                           Filter: (random() > '0'::double precision)
+                           ->  Sort
+                                 Output: t_replicate_src.i
+                                 Sort Key: t_replicate_src.i
+                                 ->  Seq Scan on rpt.t_replicate_src
+                                       Output: t_replicate_src.i
+ Optimizer: Postgres-based planner
+ Settings: enable_seqscan = 'off'
+(18 rows)
+
+insert into t_replicate_dst (i) select i from t_replicate_src;
+select distinct id from gp_dist_random('t_replicate_dst') order by id;
+ id 
+----
+  1
+  2
+  3
+  4
+  5
+(5 rows)
+
 -- update & delete
 explain (costs off) update t_replicate_volatile set a = 1 where b > random();
-ERROR:  could not devise a plan (cdbpath.c:2578)
+                     QUERY PLAN                     
+----------------------------------------------------
+ Update on t_replicate_volatile
+   ->  Seq Scan on t_replicate_volatile
+         Filter: ((b)::double precision > random())
+ Optimizer: Postgres-based planner
+(4 rows)
+
 explain (costs off) update t_replicate_volatile set a = 1 from t_replicate_volatile x where x.a + random() = t_replicate_volatile.b;
 ERROR:  could not devise a plan (cdbpath.c:2578)
 explain (costs off) update t_replicate_volatile set a = 1 from t_hashdist x where x.a + random() = t_replicate_volatile.b;
 ERROR:  could not devise a plan (cdbpath.c:2578)
 explain (costs off) delete from t_replicate_volatile where a < random();
-ERROR:  could not devise a plan (cdbpath.c:2578)
+                     QUERY PLAN                     
+----------------------------------------------------
+ Delete on t_replicate_volatile
+   ->  Seq Scan on t_replicate_volatile
+         Filter: ((a)::double precision < random())
```

Comment:
Are you sure that this plan is correct? Do we have any warranties that the same tuples will be deleted on different segments?